sycl/include/sycl/builtins.hpp

```diff
@@ -2514,89 +2614,159 @@ __SYCL_NATIVE_MATH_FUNCTION_2_OVERLOAD(powr)
+    return __sycl_std::__invoke_native_exp2<TYPE>(x);                          \
+  }
+// TODO: Replace with __SYCL_DEF_BUILTIN_GENFLOAT when merged with above
+// definition.
+__SYCL_DEF_BUILTIN_FLOAT_SCALAR
+__SYCL_DEF_BUILTIN_FLOAT_VEC
+#undef __SYCL_BUILTIN_DEF
+
+// genfloatf exp10 (genfloatf x)
+#define __SYCL_BUILTIN_DEF(TYPE)                                               \
+  inline TYPE exp10(TYPE x) __NOEXC {                                          \
+    return __sycl_std::__invoke_native_exp10<TYPE>(x);                         \
+  }
+// TODO: Replace with __SYCL_DEF_BUILTIN_GENFLOAT when merged with above
+// definition.
+__SYCL_DEF_BUILTIN_FLOAT_SCALAR
+__SYCL_DEF_BUILTIN_FLOAT_VEC
+#undef __SYCL_BUILTIN_DEF
+
+// genfloatf log (genfloatf x)
+#define __SYCL_BUILTIN_DEF(TYPE)                                               \
+  inline TYPE log(TYPE x) __NOEXC {                                            \
+    return __sycl_std::__invoke_native_log<TYPE>(x);                           \
+  }
+// TODO: Replace with __SYCL_DEF_BUILTIN_GENFLOAT when merged with above
+// definition.
+__SYCL_DEF_BUILTIN_FLOAT_SCALAR
+__SYCL_DEF_BUILTIN_FLOAT_VEC
+#undef __SYCL_BUILTIN_DEF
+
+// genfloatf log2 (genfloatf x)
+#define __SYCL_BUILTIN_DEF(TYPE)                                               \
+  inline TYPE log2(TYPE x) __NOEXC {                                           \
+    return __sycl_std::__invoke_native_log2<TYPE>(x);                          \
+  }
+// TODO: Replace with __SYCL_DEF_BUILTIN_GENFLOAT when merged with above
+// definition.
+__SYCL_DEF_BUILTIN_FLOAT_SCALAR
+__SYCL_DEF_BUILTIN_FLOAT_VEC
+#undef __SYCL_BUILTIN_DEF
+
+// genfloatf log10 (genfloatf x)
+#define __SYCL_BUILTIN_DEF(TYPE)                                               \
+  inline TYPE log10(TYPE x) __NOEXC {                                          \
+    return __sycl_std::__invoke_native_log10<TYPE>(x);                         \
+  }
+// TODO: Replace with __SYCL_DEF_BUILTIN_GENFLOAT when merged with above
+// definition.
+__SYCL_DEF_BUILTIN_FLOAT_SCALAR
+__SYCL_DEF_BUILTIN_FLOAT_VEC
+#undef __SYCL_BUILTIN_DEF
+
+// genfloatf powr (genfloatf x, genfloatf y)
+#define __SYCL_BUILTIN_DEF(TYPE)                                               \
+  inline TYPE powr(TYPE x, TYPE y) __NOEXC {                                   \
+    return __sycl_std::__invoke_native_powr<TYPE>(x, y);                       \
+  }
+// TODO: Replace with __SYCL_DEF_BUILTIN_GENFLOAT when merged with above
+// definition.
+__SYCL_DEF_BUILTIN_FLOAT_SCALAR
+__SYCL_DEF_BUILTIN_FLOAT_VEC
+#undef __SYCL_BUILTIN_DEF
+
+// genfloatf recip (genfloatf x)
+#define __SYCL_BUILTIN_DEF(TYPE)                                               \
+  inline TYPE recip(TYPE x) __NOEXC {                                          \
+    return __sycl_std::__invoke_native_recip<TYPE>(x);                         \
+  }
+// TODO: Replace with __SYCL_DEF_BUILTIN_GENFLOAT when merged with above
```

Comment:
The macro is in intel/llvm already, maybe update here?

Reply:
49ec80d